packages/lodestar-beacon-state-transition/src/fast/util/epochContext.ts

```diff
@@ -43,68 +36,74 @@ export class EpochContext {
   // Warning: may contain indices that do not yet exist in the current state, but do in a later processed state.
   public index2pubkey: PublicKey[];
   public proposers: number[];
-  public previousShuffling?: IEpochShuffling;
-  public currentShuffling?: IEpochShuffling;
-  public nextShuffling?: IEpochShuffling;
+  public previousShuffling: IEpochShuffling;
+  public currentShuffling: IEpochShuffling;
+  public nextShuffling: IEpochShuffling;
   public config: IBeaconConfig;
 
-  constructor(config: IBeaconConfig) {
-    this.config = config;
-    this.pubkey2index = new PubkeyIndexMap();
-    this.index2pubkey = [];
-    this.proposers = [];
-  }
-
-  public loadState(state: BeaconState): void {
-    this.syncPubkeys(state);
-    const currentEpoch = computeEpochAtSlot(this.config, state.slot);
+  constructor(ctx: {
+    pubkey2index: PubkeyIndexMap;
+    index2pubkey: PublicKey[];
+    proposers: number[];
+    previousShuffling: IEpochShuffling;
+    currentShuffling: IEpochShuffling;
+    nextShuffling: IEpochShuffling;
+    config: IBeaconConfig;
+  }) {
+    this.pubkey2index = ctx.pubkey2index;
+    this.index2pubkey = ctx.index2pubkey;
+    this.proposers = ctx.proposers;
+    this.previousShuffling = ctx.previousShuffling;
+    this.currentShuffling = ctx.currentShuffling;
+    this.nextShuffling = ctx.nextShuffling;
+    this.config = ctx.config;
+  }
+
```

Comment:
This approach allows to do a copy without having to edit values manually, nor use `Object.create`. At the same time by using the static method `fromState` the API is very simple to consume and makes Typescript happy as the shuffling are defined when constructing the instance.